hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/RecoverableZooKeeper.java

```diff
@@ -726,6 +726,13 @@ public synchronized States getState() {
   }
 
   public synchronized ZooKeeper getZooKeeper() {
```

Comment:
checkZk itself is synchronized, so we can remove the synchronized?

Reply:
We probably could, but I don't think this is a performance hotspot, so I'd just leave it as is.